lib/parameter_parser/search_parameter_parser.rb

```diff
@@ -267,13 +270,29 @@ def parse_dates(values)
       }
     end
 
-    def parse_date(string)
-      Date.parse(string)
+    def validate_date_key(key)
+      if !%w(from to).include?(key)
+        @errors << %{Invalid date filter parameter "#{key}:" (expected "from:" or "to:")}
```

Comment:
Do we want to allow the a result to be returned even if there are invalid keys at this point?

Reply:
We talked about this in person, but for everyone else: if `@errors` is populated, then the response is a 422 error.

So adding this line does add extra validation that wasn't there before. I think it's more useful to return an error than silently ignore the invalid filter, and it's consistent with how we treat invalid date formats. *But does anyone have any concerns with us releasing this?*